xbmc/music/MusicLibraryQueue.cpp

```diff
@@ -73,12 +77,13 @@ void CMusicLibraryQueue::ExportLibrary(const CLibExportSettings& settings, bool
 
     if (progress)
     {
-      // Render and wait for export to complete or be cancelled
+      // Render (so pointer movements work on dialog) and wait for export to 
+      // complete or be canceled
       while (progress->IsActive() && !progress->IsCanceled())
+      {
         progress->Progress();
-      // Finally close progress dialog
-      if (progress->IsActive())
-        progress->Close();
+        Sleep(10); // Arbitrary 10 millisecond wait
```

Comment:
Please, no random sleeping. The proper way to do this is to have the progress dialog set an event on cancel and wait on it here. (See GUIDialogBusy for examples)

Reply:
Sleep gone, now using event. Is thsi what you had in mind @tamland?

Reply:
Yeah, sort of. There's no need for the member variable though, it can be local for this loop (when it ends it's not active). Or even better: move this whole loop into a static method in CGUIDialogProgress such that it's reusable.

Reply:
So using the event as just a wait timer rather than something that gets set?
@tamland I had not thought to make CGUIDialogProgress something that can wait and keep rendering, but I will take a look.

Reply:
@tamland no more CEvent member variable if was unnecessary as you said  (I did get carred away). 

If the loop were a static method of `CGUIDialogProgress` what would you call it?

Reply:
`Wait`? ;)